File: fibonacci/julia/10_lucas/test_fibonacci.jl
using Test
include("fibonacci.jl")

const TESTCASES = (
    (-2, -1),
    (-1, 1),
    (0, 0),
    (1, 1),
    (2, 1),
    (3, 2),
    (10, 55),
    (40, 102334155),
    (100, 354224848179261915075),
    (250, 7896325826131730509282738943634332893686268675876375)
)

@testset "fibonacci" begin
    for (n, v) ∈ TESTCASES
        @test fibonacci(n) == v
    end
end

File: fibonacci/cxx/08_linear/test_fibonacci.cpp
#include <catch2/catch_test_macros.hpp>
#include <tuple>
#include <vector>
#include "fibonacci.hpp"

TEST_CASE("fibonacci", "test") {
  const std::vector<std::tuple<int, int>> cases{{-2, -1}, {-1, 1}, {0, 0}, {1, 1}, {2, 1}, {3, 2}, {10, 55}, {40, 102334155}};
  for (auto&& nv : cases) {
    const auto n = std::get<0>(nv);
    const auto v = std::get<1>(nv);
    REQUIRE(fibonacci(n) == v);
  }
}
